pages/selfCheckQuestions.tsx

```diff
@@ -0,0 +1,322 @@
+                "1"
+            ) {
+                lowerBound = 1;
+            }
+        }
+
+        for (let i = lowerBound; i <= upperBound; i++) {
+            optionList.push(i);
+        }
+        newList[newList.findIndex((e) => e._id === id)].options = optionList;
+        setList(newList);
+    }
+    function moveQuesitonDown(index) {
+        const newList = list.slice(0);
+        const tempQuestion = newList[index];
+        newList[index] = newList[index + 1];
+        newList[index + 1] = tempQuestion;
+        setList(newList);
+    }
+    function moveQuesitonUp(index) {
+        const newList = list.slice(0);
+        const tempQuestion = newList[index];
+        newList[index] = newList[index - 1];
+        newList[index - 1] = tempQuestion;
+        setList(newList);
+    }
+
+    function addOneQuestion(index) {
+        const newQuestion = {
+            _id: `60e642d7e4a1ae34207a92a${count + 1}`,
+            type: "multiple_choice",
+            question: "autem sunt eiusdolores nesciunt impedit?",
+            options: ["Option 1", "Option 2", "Option 3"],
+            questionNumber: 1,
+        };
+        const newList = list.slice(0);
+        const newCount = count + 1;
+        newList.splice(index + 1, 0, newQuestion);
+        setCount(newCount);
+        setList(newList);
+    }
+
+    function removeOneQuestion(id) {
+        const newList = list.filter((item) => item._id !== id);
+        setList(newList);
+    }
+    function removeAllQuestions() {
+        const newList = [];
+        setList(newList);
+    }
+
+    const selfCheckQuestionSize = list.length;
+
+    const { isOpen, onOpen, onClose } = useDisclosure();
+    return (
+        <Flex direction="column" minH="100vh">
+            <Header />
+
+            <Flex mt={10} wrap={"wrap"} justifyContent={"left"} width={"full"}>
+                <Text mx={10} fontWeight="bold" fontSize="4xl">
+                    Create a Tool
+                </Text>
+                <Spacer />
+                <ButtonGroup wrap={"wrap"} spacing={"1.5rem"} mx={12}>
+                    <Button
+                        _hover={{ bg: "#F3F3F3" }}
+                        _active={{
+                            transform: "scale(0.95)",
+                        }}
+                        onClick={onOpen}
+                        minWidth={"90"}
+                        colorScheme="white"
+                        variant="outline"
+                    >
+                        Discard
+                    </Button>
+                    <Button
+                        _hover={{ bg: "#121310" }}
+                        _active={{
+                            transform: "scale(0.95)",
+                        }}
+                        minWidth={"90"}
+                        color="white"
+                        background="black"
+                        variant="outline"
+                        onClick={() => {
```

Comment:
This is okay for now, let's keep in mind to update this when linking APIs

Reply:
We should add a TODO here so it's easier to find when searching through the codebase/if someone else picks up the ticket